### DISCUSSION TITLE
Release 1.2.0 Beta 7

### DISCUSSION THREAD

**Discussion by apohl79** [apohl79/audiogridder#940] [General]
## GENERAL
* **NEW** Use unix domain sockets for the communication between plugins, server and sandboxes:
   In the following cases we will prefer unix domain sockets over TCP for performance improvements:
  * A plugin connects to a server/sandbox with local mode enabled
  * Plugin communication with the tray application
  * Sandbox communication with the server in plugin isolation mode
* **NEW** Added IPv6 support
* **NEW** Clipboard sharing and Copy & Paste support
* Improved window positioning with local mode enabled: 
  Plugin windows are properly positioned by aligning to the DAW window size rather than the rendered editor space. This resolves issues with AG zoom enabled, if DAW plugin windows are larger than the AG editor itself and scaling on Windows.
* Improved mDNS server detection:
   The plugin will perform connectivity checks before showing a server entry. Also, remote servers in local mode will not be shown anymore.
* Fixed memory file access flags on windows, this could have led to loosing window positions 
* Fixed numpad support (resolves #938)

## PLUGIN
* **NEW** Added an option to bypass the AG plugin when disconnected or not loaded (resolves #778)
* **NEW** Added an option to keep the plugin window open on the server when closing the AG plugin in the DAW (resolves #710)
* **NEW** Allow individual buffer settings by AG plugin
* **NEW** Added tooltips to failed remote plugins that show the error string
* Improved Plugin Monitor:
  * Added "not loaded" (yellow) state and error tooltips
  * Scrolling support
* Improved AudioStreamer outbound buffering 
* Improved instrument bus/channel setup (mainly for Logic, enable more mono/stereo bus combinations)
* Fixed a memory leak in processBlock
* Fixed search window crashes
* Fixed removal of failed remote plugins

## SERVER
* **NEW** Sandbox mode - Plugin Isolation: 
  In addition to isolating a plugin chain in a separate process, each loaded remote plugin will be isolated in its own sandbox process.
* **NEW** ID based configurations:  
  Multiple servers can be run on one computer using different IDs. Each server will have a dedicated config going forward. This allows for running x86 and arm servers on the same macOS instance using different plugin folders etc.
* **NEW** Much faster plugin scans
* **NEW** Multi-select actions (activate/deactivate/unban) in the plugin manager (resolves #892)
* Try a few more channel layouts before falling back to the plugin default
* Improved plugin ID (VST3) cross platform compatibility
* Improved plugin UI bounds detection
* Make sure plugin windows get positioned in the user area to avoid overlapping taskbars (resolves #671)
* Show a warning with plugins that failed after scanning
* Fixed a VST3 initialization bug
* Fixed mouse position calculation in fullscreen mode
* Fixed some memory corruption errors

<hr /><em>This discussion was created from the release <a href='https://github.com/apohl79/audiogridder/releases/tag/release_1_2_0_beta7'>Release 1.2.0 Beta 7</a>.</em>

**Comment by Pabloscotty** [comment]
Thank You Andreas!!!

**Reply by Fittkenny** [reply]
Sweeeeet!

**Comment by emerymastering** [comment]
Ohh, nice, lets try!) 
BBigg ups Andreas!!

**Comment by djschultz123** [comment]
I had to revert to v6 on my Win10 computer in Studio One Pro 5.latest,

With v7, the plugin window anchored in one place and also the Audiogridder plugin and the synth plugin it loaded were also locked into relative position to one another. Very weird.

Worse, I couldn't use a menu in the softsynth loaded into Audiogridder because the menu kept automatically closing as I tried to use it. 

I'll try again with the next version.

**Reply by apohl79** [reply]
Sounds weird. I assume you have the local mode enabled. I tested this with the S1 5 demo version and things look good. The server plugin window attaches itself to the AG plugin window in S1. Maybe, when you get a chance, you could provide a short screen recording showing the issue. Cheers.

**Reply by jaacMusic** [reply]
I'm experiencing the same issue when trying to load a plugin. The dropdown menu listing keeps closing when I'm about to click on a plugin. It happens randomly; need to conduct more testing!

Thanks!

**Comment by emack999** [comment]
Thanks for the update.
Windows version, running in Local mode--connection between the AG plugin and server gets interrupted.  Have never had connection issues in any other version before.  So, I isolated it down to an issue with IPv4 and IPv6 running at the same time on the loopback adapter.  Once I disabled IPv4, the connection issue resolved itself, with no further disconnections.  You should probably should have a look at that though.

Also, when loading a plugin, the plugin window appears.  If I 'x' it out, it just reappears a few seconds later.  I have to click on the plugin name in the AG window to 'un-highlight' it, and then the plugin window stays closed.  Don't know if this is intentional, but I'm just noting it because it is different behavior from v6. 

I'm also not able to reposition a loaded plugin window without it returning to the original position.  For instance, I can move a plugin window to a 2nd monitor, but it returns to the 1st monitor almost instantly.

**Reply by apohl79** [reply]
great stuff, I realized some issues in that area a few mins ago, but intended to continue debugging tomorrow. thx for this info.

**Reply by apohl79** [reply]
Thanks for the info last night. Speeded up my analysis a lot. This is fixed in beta8.

**Comment by tbffbt** [comment]
I realize I'm not exactly the main target audience, this being a beta version. but my somewhat obsolete system (Mojave 10.14.6 and Ableton 11.0.11) only runs into issues with the main FX instance, with the same old constant CPU spikes of over 1000% from just loading AG. both MIDI and Instrument instances didn't cause spiking. hope this is useful

**Reply by AmnesiaJones** [reply]
I'm using the same OS and DAW versions as you. Stable with no issues, running in local mode. What buffer size are you set to in audiogridder?
